yb-voyager/src/queryparser/helpers.go

```diff
@@ -0,0 +1,100 @@
+package queryparser
+
+import (
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+var unsupportedAdvLockFuncs = []string{
+	"pg_advisory_lock", "pg_try_advisory_lock", "pg_advisory_xact_lock",
+	"pg_advisory_unlock", "pg_advisory_unlock_all", "pg_try_advisory_xact_lock",
+}
+
+var unsupportedSysCols = []string{
+	"xmin", "xmax", "cmin", "cmax", "ctid",
+}
+
+// TODO: verify the list (any missing, or extra/invalid)
+var xmlFunctions = []string{
+	"cursor_to_xml", "cursor_to_xmlschema", // Cursor to XML
+	"database_to_xml", "database_to_xml_and_xmlschema", "database_to_xmlschema", // Database to XML
+	"query_to_xml", "query_to_xml_and_xmlschema", "query_to_xmlschema", // Query to XML
+	"schema_to_xml", "schema_to_xml_and_xmlschema", "schema_to_xmlschema", // Schema to XML
+	"table_to_xml", "table_to_xml_and_xmlschema", "table_to_xmlschema", // Table to XML
+	"xmlagg", "xmlcomment", "xmlconcat2", // XML Aggregation and Construction
+	"xmlexists", "xmlvalidate", // XML Existence and Validation
+	"xpath", "xpath_exists", // XPath Functions
+	"xml_in", "xml_out", "xml_recv", "xml_send", // System XML I/O
+	"xml", // Data Type Conversion
+}
+
+// Sample example: {func_call:{funcname:{string:{sval:"pg_advisory_lock"}}
+func getFuncNameFromFuncCall(funcCallNode protoreflect.Message) string {
+	funcnameField := funcCallNode.Get(funcCallNode.Descriptor().Fields().ByName("funcname"))
```

Comment:
You should check here in this function that the node type is matching what you expect. (a PG_QUERY_FUNCCALL_NODE)

Reply:
this helper func is being called from FuncCallDetector.Detect(). where we have check for the nodeType.
If i add here it will be redundant.

Reply:
Right, but still, as a standalone function that lets you `getFuncNameFromFuncCall`, I think it makes sense for the function to validate. 

Worst case you might hit a panic.
Which just made me realise - I guess it's possible for `funcnameField` to be NULL, right? We should add that check before accessing it. (same for the columnref func)

Reply:
It cant be null
```
func (protoreflect.Message) Get(protoreflect.FieldDescriptor) protoreflect.Value
Get retrieves the value for a field.

For unpopulated scalars, it returns the default value, where the default value of a bytes scalar is guaranteed to be a copy. For unpopulated composite types, it returns an empty, read-only view of the value; to obtain a mutable reference, use Mutable.
```